src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,163 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    Node<T> node(int index) {
+        Node<T> find;
+        if (index < (size / 2)) {
+            find = head;
+            for (int i = 0; i < index; i++) {
+                find = find.next;
+            }
+        } else {
+            find = tail;
+            for (int i = size - 1; i > index; i--) {
+                find = find.prev;
+            }
+        }
+        return find;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> t = tail;
+        Node<T> newNode = new Node<>(t, value, null);
+        tail = newNode;
+        if (t == null) {
+            head = newNode;
+        } else {
+            t.next = newNode;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
 
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> actual = node(index).prev;
+            Node<T> newNode = new Node<>(actual, value, node(index));
```

Comment:
```suggestion
            Node<T> newNode = new Node<>(actual, value, actual.next);
```
Нет смысла вызывать там метод по поиску ноды, можно сразу обратиться к следующему узлу ноды actual и передать его в качестве параметра.